server/channels/api4/group.go

```diff
@@ -631,15 +625,47 @@ func unlinkGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func verifyLinkUnlinkPermission(c *Context, syncableType model.GroupSyncableType, syncableID string) *model.AppError {
+	group, appErr := c.App.GetGroup(c.Params.GroupId, nil, nil)
+	if appErr != nil {
+		return appErr
+	}
+
+	if group.Source != model.GroupSourceLdap {
+		return model.NewAppError("Api4.linkGroupSyncable", "app.group.crud_permission", nil, "", http.StatusBadRequest)
+	}
+
+	// If AllowReference is disabled, limit who can link the group.
+	// This voids leaking the list of group members.
+	// See https://mattermost.atlassian.net/browse/MM-55314 for more details.
+	if !group.AllowReference {
+		if !c.App.SessionHasPermissionToGroup(*c.AppContext.Session(), c.Params.GroupId, model.PermissionSysconsoleReadUserManagementGroups) {
```

Comment:
I know we had said that we need to check for `PermissionSysconsoleReadUserManagementGroups` but I wonder if the `PermissionSysconsoleWriteUserManagementGroups` permission is more appropriate here as this is essentially an operation that changes the state of the group and as such Write is more appropriate than Read.

Reply:
Fair point, updated the required permission to `PermissionSysconsoleWriteUserManagementGroups` via aefe679a7d.

Reply:
Actually, I wonder if the changes in https://github.com/mattermost/mattermost/commit/aefe679a7d19e02e91cc66479f076f90c2c70012 is incorrect given the statement from https://github.com/mattermost/mattermost/pull/26595#discussion_r1555611339:

> I see that linking a group would count as modifying it. But this check protects the member list, not the state of the group.

Reply:
I'd prefer it `PermissionSysconsoleWriteUserManagementGroups` but I'm ok if you want to revert it to Read to keep the change specific to the ticket and have a new ticket for this.